flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java

```diff
@@ -326,4 +327,51 @@ public void testTwoSinksInDisjointedDAG() throws Exception {
     Assert.assertEquals("rightTable", rightTable.currentSnapshot().summary().get("direction"));
   }
 
+  @Test
+  public void testOverrideWriteConfigWithUnknownDistributionMode() {
+    AssertHelpers.assertThrows("Override write configuration.",
+            RuntimeException.class, "No enum constant org.apache.iceberg.DistributionMode.UNRECOGNIZED",
```

Comment:
RuntimeException is too broad. Can you test a more specific exception? Otherwise anything that goes wrong in this huge block looks like the test is passing.

Reply:
Replaced with IllegalArgumentException.